app/controllers/triannon/application_controller.rb

```diff
@@ -39,13 +44,90 @@ def access_token_valid?(code)
     # @return login_data [Hash|nil]
     def access_token_data(headers)
       auth = headers['Authorization']
-      if auth.nil? || auth !~ /Bearer/
-        nil
-      else
-        token = auth.split[1]
+      unless auth.nil? || auth !~ /^Bearer/
+        token = auth.split.last
         access_token_valid?(token)
       end
     end
 
+
+    private
+
+    def authorize
+      # Require authorization on POST and DELETE requests.
+      auth_methods = ['POST','DELETE']
+      return true unless auth_methods.include? request.method
+      # Allow any requests to the /auth paths; provided that an
+      # anno root container cannot start with 'auth' in the name
+      # (which is controlled by the routes constraints).
+      return true if request.path =~ /^\/auth/
+      authorized_workgroup?
+    end
+
+    def authorized_workgroup?
+      # If the request does not map to a configured container, allow access.
+      container_auth = container_authorization
+      return true if container_auth.empty?
+      if request.headers['Authorization'].nil?
+        render401
+        return false
+      end
+      # Identify an intersection of the user and the authorized workgroups.
+      container_groups = container_auth['workgroups'] || []
+      match = container_groups & user_workgroups
+      if match.empty?
+        render403
+        false
+      else
+        true
+      end
+    end
+
+    # Extract container authorization from the configuration parameters
+    # @return authorization [Hash]
+    def container_authorization
+      container_config = request_container_config
+      if container_config.instance_of? Hash
+        container_config['auth'] || {}
+      else
+        {}
+      end
+    end
+
+    # Map the request root container to config parameters;
+    # assume that a request can only map to one root container.
+    # If this mapping fails, assume that authorization is OK.
+    def request_container_config
+      # TODO: refine the matching algorithm, esp if there is more than one
+      # match rather than assume it can only match one container.
+      request_container = params['anno_root'] || request.path
```

Comment:
Simplify to:
```
Triannon.config[:ldp]['anno_containers'][params['anno_root']]
```
Meaning... treat the anno_root as a key without trying to match beyond that. If it doesn't exist then it'll return nil, as expected?